jmx-scraper/src/main/java/io/opentelemetry/contrib/jmxscraper/config/JmxScraperConfigFactory.java

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.jmxscraper.config;
+
+import static io.opentelemetry.contrib.jmxscraper.util.StringUtils.isBlank;
+
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+@SuppressWarnings({"SystemOut", "SystemExitOutsideMain"})
+public class JmxScraperConfigFactory {
+  private static final String PREFIX = "otel.";
+  static final String SERVICE_URL = PREFIX + "jmx.service.url";
+  static final String CUSTOM_JMX_SCRAPING_CONFIG = PREFIX + "jmx.custom.jmx.scraping.config";
+  static final String TARGET_SYSTEM = PREFIX + "jmx.target.system";
+  static final String INTERVAL_MILLISECONDS = PREFIX + "jmx.interval.milliseconds";
+  static final String METRICS_EXPORTER_TYPE = PREFIX + "metrics.exporter";
+  static final String EXPORTER_INTERVAL = PREFIX + "metric.export.interval";
+  static final String REGISTRY_SSL = PREFIX + "jmx.remote.registry.ssl";
+
+  static final String OTLP_ENDPOINT = PREFIX + "exporter.otlp.endpoint";
+
+  static final String JMX_USERNAME = PREFIX + "jmx.username";
+  static final String JMX_PASSWORD = PREFIX + "jmx.password";
+  static final String JMX_REMOTE_PROFILE = PREFIX + "jmx.remote.profile";
+  static final String JMX_REALM = PREFIX + "jmx.realm";
+
+  // These properties need to be copied into System Properties if provided via the property
+  // file so that they are available to the JMX Connection builder
+  static final List<String> JAVA_SYSTEM_PROPERTIES =
+      Arrays.asList(
+          "javax.net.ssl.keyStore",
+          "javax.net.ssl.keyStorePassword",
+          "javax.net.ssl.keyStoreType",
+          "javax.net.ssl.trustStore",
+          "javax.net.ssl.trustStorePassword",
+          "javax.net.ssl.trustStoreType");
+
+  static final List<String> AVAILABLE_TARGET_SYSTEMS =
+      Arrays.asList(
+          "activemq",
+          "cassandra",
+          "hbase",
+          "hadoop",
+          "jetty",
+          "jvm",
+          "kafka",
+          "kafka-consumer",
+          "kafka-producer",
+          "solr",
+          "tomcat",
+          "wildfly");
+
+  private Properties properties = new Properties();
+
+  /**
+   * Create {@link JmxScraperConfig} object basing on command line options
+   *
+   * @param args application commandline arguments
+   */
+  public JmxScraperConfig createConfigFromArgs(List<String> args) {
+    if (!args.isEmpty() && (args.size() != 2 || !args.get(0).equalsIgnoreCase("-config"))) {
+      System.out.println(
+          "Usage: java io.opentelemetry.contrib.jmxscraper.JmxScraper "
```

Comment:
The actual command is based on `java -jar` as the main class is provided in the manifest. If we really want to have the original JVM command line we can get it through JMX, but I don't think it's worth the extra complexity.